#1689 - Fix vertical order book centering

kapeer42 opened the pull request:
Closes #1689

I took the liberty to rename `OrderRow` into `OrderRows` since it's multiple rows, was confusing me :P

sschiessl-bcp commented:
Did you do this fix for the old exchange layout, or the new one? Or does it apply to both as the OrderBook component was unaffected?

startailcoon commented:
As you made this based on the dev branch with Issue 1477 within it, I decided to try it out with it to see how it works.

It works much better, but it seems that there is still some issues with the centering sometimes. Same thing happens both when it's locked as well as unlocked.

![bitshares-autoscroll_on](https://user-images.githubusercontent.com/12114550/45773965-0a3b6080-bc4c-11e8-9104-8afc85de6672.gif)

startailcoon commented on app/components/Exchange/OrderBook.jsx:
```diff
@@ -427,49 +424,31 @@ class OrderBook extends React.Component {
 
-    componentWillUpdate() {
+    getSnapshotBeforeUpdate() {
         if (!this.props.horizontal) {
```
To make the user able to disable auto scroll, this has to be `if (!this.props.horizontal && this.state.autoScroll) {`

kapeer42 commented:
I updated to latest develop, reworked it and did a `push --force`.

I updated the lifecycle hooks to latest react.

There is still is a slight delay in some cases but I spent way too many time on this already.

The `verticalStickyTable.table` reference seems to be outdated in `OrderBook.componentDidUpdate`.
Even with a direct `document.querySelector("#top-order-rows").getBoundingClientRect().height` I get bogus values for the asks group height. Had to compute it from the number of asks times one ask height.

sschiessl-bcp commented:
I can also see the improvement.

It might be a good idea to change that in the future to two components and two scrollbars. I see little value in a moving Latest/Spread column. We will see.
